share/osutil/file_linux.go

```diff
@@ -45,48 +45,87 @@ type FileInfoExt struct {
 }
 
 
-func fileExists(path string) bool {
-    _, err := os.Lstat(path)
-    return !errors.Is(err, os.ErrNotExist)
+func fileExists(path string) (err error) {
+    _, err = os.Lstat(path)
+    if err == nil {
+        // No error, file exists and is accessible
+        return nil
+    }
+
+    if errors.Is(err, os.ErrNotExist) {
+        // File does not exist
+        return err
+    }
+
+    if os.IsPermission(err) {
+        return fmt.Errorf("Permission error accessing file: %s", err)
+    }
+
+    // Other types of errors
+    return err
```

Comment:
Sorry if I didn't explain clearly in my last comment.  You can use one of below patterns to implement a function that would return error occasionally:
1. Return `(bool, error)`.  It becomes caller's responsibility to handle the error.
2. Still return `bool` but prints log when an error happens.